### PR TITLE
Add @bull-board/ui as peerDep of @bull-board/api

### PR DESCRIPTION
**Pull request by JasonMan34**
This closes https://github.com/felixmosh/bull-board/issues/564

### CODE REVIEW

File: packages/api/package.json [hunk 1]
```diff
@@ -36,6 +36,9 @@
     "bullmq": "^1.80.6",
     "ioredis-mock": "^7.2.0"
   },
+  "peerDependencies": {
+    "@bull-board/ui": "5.1.1"
```
Comment (felixmosh) [inline packages/api/package.json:40]: lets not restrict the version here, make it `*`

Comment (JasonMan34) [reply packages/api/package.json]: Wasn't sure about compatibility issues so I copied the version from the inverse dependency (ui's dependany on api@5.1.1)

Changed to *

**Comment by felixmosh** [comment]
Released in `v5.1.2`, can you try this version?

**Comment by JasonMan34** [comment]
Works now :)

Although the solution feels off to me. `ui`'s dependency of `api` is obvious, but the reverse dependency only exists because of this piece of code:
```ts
export function createBullBoard({
  queues,
  serverAdapter,
  options = { uiConfig: {} },
}: {
  queues: ReadonlyArray<BaseAdapter>;
  serverAdapter: IServerAdapter;
  options?: BoardOptions;
}) {
  const { bullBoardQueues, setQueues, replaceQueues, addQueue, removeQueue } = getQueuesApi(queues);
  const uiBasePath = path.dirname(eval(`require.resolve('@bull-board/ui/package.json')`));

  serverAdapter
    .setQueues(bullBoardQueues)
    .setViewsPath(path.join(uiBasePath, 'dist'))
    .setStaticPath('/static', path.join(uiBasePath, 'dist/static'))
    .setUIConfig(options.uiConfig)
    .setEntryRoute(appRoutes.entryPoint)
    .setErrorHandler(errorHandler)
    .setApiRoutes(appRoutes.api);

  return { setQueues, replaceQueues, addQueue, removeQueue };
}
```

Perhaps if `createBullBoard` is inherently coupled with `@bull-board/ui`, it should be exported from there? That would remove the cyclic dependency and resolve the issue (without declaring a peerDep)  
Alternatively, perhaps `createBullBoard` should be decoupled from `@bull-board/ui` entirely, and accept another required parameter `uiBasePath`. Then `@bull-board/ui` could export a basePath to be used when calling `createBullBoard`

**Comment by felixmosh** [comment]
Thanx for the feedback.

1. `createBullBoard` is a builder of api, it should be on the server side part.
2. I've thought about it, but then each user of the lib will need to pass the same piece of code
3. Good possible solution, I'll consider this refactor, thanx

**Pull request merged**